### ISSUE TITLE
[Bug]: Problems running Vitest accessibility testing in storybook/html-vite

### ISSUE THREAD

**Issue by basher** [storybooks/storybook#34613]
### Describe the bug

I've installed Vitest accessibility testing to my Storybook/HTML component library.

1. I am getting (spurious) accessibility errors reported by Vitest in the Storybook UI, but with no explicit details about the errors.
2. The Vitest CLI reports errors with AxeCore itself.

I have setup a minimal repo with a single Button component to demonstrate the issue. See reproduction link below.

> NOTE: I notice that Storybook/HTML (with Vite) is not officially supported yet for Vitest - https://storybook.js.org/docs/writing-tests/integrations/vitest-addon

### Reproduction link

https://github.com/basher/storybook-html-vitest-a11y

### Reproduction steps

Please see steps (and screenshots of the issues) in the reproduction repo `README`

### System

```bash
Storybook Environment Info:
│

│  System:
│  OS: Windows 10 10.0.19045
│  CPU: (4) x64 Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz
│  Binaries:
│  Node: 22.19.0 - C:\Program Files\nodejs\node.EXE
│  Yarn: 1.22.22 - C:\Users\kev\AppData\Roaming\npm\yarn.CMD
│  npm: 10.9.3 - C:\Program Files\nodejs\npm.CMD <----- active
│  Browsers:
│  Chrome: 147.0.7727.102
│  Edge: Chromium (140.0.3485.94)
│  Firefox: 150.0 - C:\Program Files (x86)\Mozilla Firefox\firefox.exe
│  npmPackages:
│  @storybook/addon-a11y: ^10.3.5 => 10.3.5
│  @storybook/addon-docs: ^10.3.5 => 10.3.5
│  @storybook/addon-vitest: ^10.3.5 => 10.3.5
│  @storybook/html-vite: ^10.3.5 => 10.3.5
│  storybook: ^10.3.5 => 10.3.5
```

**Comment by basher** [comment]
@dosu
To answer your questions/suggestions:

1. **Spurious errors in UI**. My Button component does not have any animations, etc. Just some simple static styles that define foreground, background and border colors. See screenshot below - no details in the UI when Vitest accessibility tests fail:

<img width="2077" height="577" alt="Image" src="https://github.com/user-attachments/assets/4d62eb73-4f5e-413b-aca4-2e8efd432884" />


2. **AxeCore CLI errors**. I've added a new [vitest.setup.js](https://github.com/basher/storybook-html-vitest-a11y/blob/master/.storybook/vitest.setup.js) file. See screenshot below for CLI errors:

<img width="1630" height="1118" alt="Image" src="https://github.com/user-attachments/assets/1a26f007-f991-4563-9d8a-2eb0ec1cef89" />


3. **Windows file path**. I'd already seen other issues reported about this, and have accounted for it in my [vitest.config.js](https://github.com/basher/storybook-html-vitest-a11y/blob/master/vitest.config.js#L7).

**Comment by basher** [comment]
@dosu
The `@storybook/addon-vitest` is there inside my `main.js`.
See https://github.com/basher/storybook-html-vitest-a11y/blob/22a574365c9b03f76a68b5a82355eeb15aee18b8/.storybook/main.js#L9

Otherwise I would not have been able to run Vitest from the UI.

**Comment by basher** [comment]
I appreciate that I'm trying Vitest in an unsupported framework... hopefully registering my issues now may help others, and even help with testing Storybook/ HTML when that framework is added to Vitest support.

**Comment by basher** [comment]
@dosu
I've already added the CLI errors screenshot in a previous comment, but I'll add it again for you here:

<img width="1630" height="1118" alt="Image" src="https://github.com/user-attachments/assets/ac7cea76-83c7-4075-885f-bb89939e50c9" />

**Comment by basher** [comment]
Thanks @dosu

I now have passing tests in both the UI and CLI, as a result of changing my Vitest `context`.
See commit https://github.com/basher/storybook-html-vitest-a11y/commit/05327632e72583f97ca723800d1b07d2d5e7d628

I've also updated my imports in `vitest.setup.js` as per your suggestion.

Tomorrow, I will implement these changes in my [proper UI boilerplate repo](https://github.com/basher/Web-UI-Boilerplate) and run the tests again.

**Comment by basher** [comment]
@dosu

I've run Vitest multiple times in my [UI boilerplate repo](https://github.com/basher/Web-UI-Boilerplate)

1. **Via Storybook UI (Firefox and Chrome)**

- First run, Vitest *hangs* on component test 88/96 without any errors in UI or CLI.
- Subsequent runs (after clearing browser cache and restarting Storybook server), connection was lost in UI, and got following error in CLI:

```
storybook/test: TypeError: No existing
│  state found for follower with id:
│  'storybook/status'. Make sure a leader
│  with the same id exists before creating
│  a follower.
│  at Timeout._onTimeout
│  (file:///C:/Users/kev/Documents/work/Projects/Web-UI-Boilerplate/ui/node_modules/storybook/dist/core-server/index.js:6834:9)
│  at listOnTimeout
│  (node:internal/timers:588:17)
│  at process.processTimers
│  (node:internal/timers:523:7)
```

2. **Via CLI**
- Multiple runs, which all *hung* after 6 passed tests:

```
$ npm run test-storybook

> web-ui-boilerplate@6.4.3 test-storybook 
> vitest --project=storybook --run --reporter verbose

 RUN  v4.1.4 C:/.../Web-UI-Boilerplate/ui

 ✓  storybook (chromium)  stories/3. Layout/Flex/Flex.stories.js > Flex 661ms
 ✓  storybook (chromium)  stories/3. Layout/Wrapper/Wrapper.stories.js > Wrapper 415ms
 ✓  storybook (chromium)  stories/3. Layout/MarginPadding/MarginPadding.stories.js > Adjust Margin & Padding 500ms
 ✓  storybook (chromium)  stories/5. Components/Cards/Cards.stories.js > Card 1335ms
 ✓  storybook (chromium)  stories/5. Components/Cards/Cards.stories.js > Fullwidth Card 130ms
 ✓  storybook (chromium)  stories/5. Components/Cards/Cards.stories.js > Clickable Card 687ms

 Test Files 4 passed (60)
      Tests 6 passed (6)
   Start at 10:00:54
   Duration 244.14s
```

**Comment by basher** [comment]
@dosu

I have implemented your suggested fixes in commit https://github.com/basher/Web-UI-Boilerplate/commit/1c5fdcbf313a1e29c9f4b585a522c2efdf943f27

> NOTE: I have deleted `vitest.setup.js` as it makes no difference to the outcome. I already had `@storybook/addon-a11y` installed in my repo before I installed `@storybook/addon-vitest`.

- All 96 tests complete in the UI, but only in Chrome. In Firefox, I still get a timeout error in the UI, but the tests appear to complete when I look at the CLI.
- All 96 tests complete in the CLI.

-------------------------

My next step is to integrate this into my CI/CD pipleline.

**Comment by basher** [comment]
@dosu
Further to my previous comment...

Vitest completes all 96 tests in my [UI boilerplate repo](https://github.com/basher/Web-UI-Boilerplate) CLI, but **some of the tests are failing**:

- I am unable to [disable `target-size` tests via a config rule](https://github.com/basher/Web-UI-Boilerplate/blob/dc07a5e61e89d3d5c740a51c1aa855fba0b43054/ui/.storybook/preview.js#L35), which are wrongly failing. In the UI, these `target-size` tests pass. I've also tested these (failing) components standalone with 3 different automated accessibility browser plugins (including Axe), so I know the **CLI error is a false negative**.
- I have [removed `best-practice` from the `runOnly` option](https://github.com/basher/Web-UI-Boilerplate/blob/dc07a5e61e89d3d5c740a51c1aa855fba0b43054/ui/.storybook/preview.js#L43) as the Axe `region` rule fails for most of my components, and I am unable to disable this test via the config.

However, in [my minimal repo](https://github.com/basher/storybook-html-vitest-a11y) CLI:
- `target-size` test passes for the small button variant. The HTML/CSS for this component is exactly the same as per my UI boilerplate.
- [`best-practice` tests have not been removed](https://github.com/basher/storybook-html-vitest-a11y/blob/8e2f85f63767c64a4b37411fb081747f58e1636b/.storybook/preview.js#L12), and I get no Axe `region` rule failures in the CLI.

I'm confused as to why the same component fails in one repo, but not the other???

Both repos are running the latest versions of `vitest` and `@vitest/browser-playwright`, and I disable cache with `--no-cache`.

**Comment by basher** [comment]
@dosu

- I've re-instated `vitest.setup.js` in UI boilerplate.
- I have no story-level a11y overrides.
- I temporarily commented out the decorator. This decorator only adds a CSS `<link>` statement in **PRODUCTION** environment, allowing me to use a theme switcher to swap out different theme CSS files in the published Storybook UI.
- I temporarily added `console.log` to a failing story, as you suggested, and I got the following output, showing the config being applied:
```
a11y config: {"test":"error","context":"body","config":{"rules":[{"id":"target-size","enabled":false}]},"options":{"runOnly":["wcag2a","wcag2aa","wcag21a","wcag21aa","wcag22a","wcag22aa"]}}
```

The issues still persist.

**Comment by basher** [comment]
@dosu

I have created a new temporary `storiesCopy` folder in the boilerplate repo, and have tested a number of components both individually, and grouped together in small batches.

I have **made no changes to any of the Storybook or Vitest configs / setup**, other than temporarily pointing `main.js` to the new `storiesCopy` folder:

```
const config = {
    stories: [
        '../storiesCopy/**/*.mdx',
        '../storiesCopy/**/*.stories.@(js|jsx|ts|tsx)'
    ],
```

Here are some results of my testing:

- [Buttons stories](https://github.com/basher/Web-UI-Boilerplate/tree/master/ui/stories/5.%20Components/Buttons%20%26%20Links/Buttons) **failed** `target-size` test (e.g. when `buttonSize` arg is set to `small`) and `region` test when all 96 tests are run, but **passed** both tests when tested in isolation.
- [Accordion stories](https://github.com/basher/Web-UI-Boilerplate/tree/master/ui/stories/5.%20Components/Accordion) **failed** `target-size` and `region` tests when all 96 tests are run, and still **failed** both tests when tested in isolation.
- [Form stories](https://github.com/basher/Web-UI-Boilerplate/tree/master/ui/stories/4.%20Forms/Form) **failed** `target-size` and `region` tests when all 96 tests are run, and still **failed** both tests when tested in isolation.

So there is a confusing mix of results... possibly indicating both an issue with state leakage, and environment (i.e. Storybook/HTML)???

**Comment by basher** [comment]
Thanks @dosu - that worked!

I reverted `main.js` to re-run all 96 tests, and they **all passed** too.

So it does appear we've identified a bug in the addon?

Additionally:
1. I shouldn't need to disable `region` rule because, as you stated in an earlier comment, `addon-a11y` actually disables it by default because isolated components typically lack full-page landmarks.
2. I shouldn't need to disable `target-size` because the Storybook UI (and 3 different accessibility devtools plugins) all pass, and I want to be able to identify genuine failures if they exist, not ignore/bypass them.

**Comment by basher** [comment]
@dosu - I've registered, so hopefully we can continue...?

**Comment by basher** [comment]
@dosu

Yes, I did try the workaround - see [my comment above, with additional questions](https://github.com/storybookjs/storybook/issues/34613#issuecomment-4334413133).

**Comment by basher** [comment]
Thanks @dosu - I will now close this issue, and create a new one specifically for `addon-a11y` config issues.
